app/Main.hs

```diff
@@ -62,6 +64,18 @@ hsOutP =
     optional $ strOption $
     long "hs-out" <> metavar "DIR" <> help "Generate Haskell bindings to specified dir"
 
+elmVersionP :: Parser (Maybe ElmVersion)
+elmVersionP =
+    mkElmVersion <$>
+    (optional $ strOption $
```

Comment:
Well, currently if you supply an unsupported elm version it will automatically (silently) fallback to 0.16. If you use eitherReader you can output an error through optparse-applicative explaining what went wrong. Why not derive `Enum`/`Bounded`?